csharp/src/Google.Protobuf/Reflection/SingleFieldAccessor.cs

```diff
@@ -109,14 +113,50 @@ internal SingleFieldAccessor(PropertyInfo property, FieldDescriptor descriptor)
                 // While presence isn't supported, clearing still is; it's just setting to a default value.
                 var clrType = property.PropertyType;
 
-                object defaultValue =
-                    clrType == typeof(string) ? ""
-                    : clrType == typeof(ByteString) ? ByteString.Empty
-                    : Activator.CreateInstance(clrType);
+                object defaultValue = GetDefaultValue(descriptor);
```

Comment:
This is the one notable runtime change. A default value is returned based on the field type instead of using reflection.